src/toast/toaster.js

```diff
@@ -0,0 +1,235 @@
+  };
+
+  show = (props: $Shape<ToastPropsT> = {}): React.Key => {
+    const toastProps = this.getToastProps(props);
+    this.setState(({toasts}) => {
+      toasts.unshift(toastProps);
+      return {toasts};
+    });
+    return toastProps.key;
+  };
+
+  update = (key: React.Key, props: ToastPropsT): void => {
+    this.setState(({toasts}) => {
+      toasts.forEach((t, index, arr) => {
+        if (t.key === key) {
+          arr[index] = {...t, ...this.getToastProps(props), key};
+        }
+      });
+      return {
+        toasts,
+      };
+    });
+  };
+
+  dismiss = (key: React.Key) => {
+    if (this.dismissHandlers[key]) {
+      this.dismissHandlers[key]();
+    }
+  };
+
+  clearAll = () => {
+    Object.keys(this.dismissHandlers).forEach(key => {
+      this.dismissHandlers[key]();
+    });
+  };
+
+  clear = (key?: React.Key) => {
+    key === undefined ? this.clearAll() : this.dismiss(key);
+  };
+
+  internalOnClose = (key: React.Key) => {
+    delete this.dismissHandlers[key];
+    this.setState(({toasts}) => ({
+      toasts: toasts.filter(t => {
+        return !(t.key === key);
+      }),
+    }));
+  };
+
+  getOnCloseHandler = (key: React.Key, onClose: ?() => void) => {
+    return () => {
+      this.internalOnClose(key);
+      typeof onClose === 'function' && onClose();
+    };
+  };
+
+  renderToast = (toastProps: ToastPropsT & {key: React.Key}): React.Node => {
+    const {onClose, children, key, ...rest} = toastProps;
+
+    const {
+      ToastBody: BodyOverride,
+      ToastCloseIcon: CloseIconOverride,
+    } = this.props.overrides;
+    const globalToastOverrides = mergeOverrides(
+      {Body: StyledBody, CloseIcon: StyledCloseIcon},
+      // $FlowFixMe
+      {Body: BodyOverride, CloseIcon: CloseIconOverride},
+    );
+    const toastOverrides = mergeOverrides(
+      globalToastOverrides,
+      toastProps.overrides,
+    );
+
+    return (
+      <Toast
+        {...rest}
+        overrides={toastOverrides}
+        key={key}
+        onClose={this.getOnCloseHandler(key, onClose)}
+      >
+        {({dismiss}) => {
+          this.dismissHandlers[key] = dismiss;
+          return children;
+        }}
+      </Toast>
+    );
+  };
+
+  getSharedProps = () => {
+    const {placement} = this.props;
+    return {
+      $placement: placement,
+    };
+  };
+
+  render = () => {
+    const sharedProps = this.getSharedProps();
+
+    const {Root: RootOverride} = this.props.overrides;
+    // $FlowFixMe
+    const [Root, rootProps] = getOverrides(RootOverride, StyledRoot);
+
+    const root = (
+      <Root {...sharedProps} {...rootProps}>
+        {this.state.toasts.map(this.renderToast)}
+      </Root>
+    );
+    if (this.state.isMounted) {
+      // Only render on the browser (portals aren't supported server-side)
+      if (this.props.usePortal) {
+        if (__BROWSER__) {
+          return ReactDOM.createPortal(
+            root,
+            // $FlowFixMe
+            document.body,
```

Comment:
> As I pointed we may need to add optional parent Node to be used instead of body.
I considered it, though decided not to add it at this point since the container is positioned relative to a browser window. We can always add it if there is a need for it.

> As for fixed elements there will be issue to adjust toasts to not overlap
Could you clarify what you mean here? The toasts are not overlapped neither positioned `fixed`. Only the toaster container has `fixed` position.

Reply:
do we see navigation bar?
http://jsfiddle.net/dj8avtkn/2/

Reply:
The multiple toasters issue had been described in the [docs](https://github.com/uber-web/baseui/pull/370/files#diff-b4255f27ec32bc0f807bf1ed86a9cbbaR135). Were you referring to the same "multiple toasters" issue in your question?

Reply:
Currently there is an option `usePortal` that can be set to `false` in order to have the toaser rendered in a different element. The fixed positioning styles would still be applied to the toasters container element, but those are overrideable through the regular `overrides` way.

Reply:
ok